executors/keptn/pkg/actions/install.go

```diff
@@ -0,0 +1,103 @@
+package actions
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/Azure/orkestra-workflow-executor/executors/keptn/pkg/keptn"
+	"github.com/Azure/orkestra-workflow-executor/executors/keptn/pkg/status"
+	fluxhelmv2beta1 "github.com/fluxcd/helm-controller/api/v2beta1"
+)
+
+func Install(ctx context.Context, cancel context.CancelFunc, hr *fluxhelmv2beta1.HelmRelease, interval time.Duration, data map[string]string) error {
+	keptnConfig := &keptn.KeptnConfig{}
+
+	// Read the keptn-config.yaml file.
+	// This file is required and cannot have empty fields
+	if v, ok := data[keptn.KeptnConfigFileName]; !ok {
+		return fmt.Errorf("failed to read plugin configuration from configmap")
+	} else {
+		if err := json.Unmarshal([]byte(v), keptnConfig); err != nil {
+			return fmt.Errorf("failed to unmarshal the keptn configuration file into KeptnConfig object")
+		}
+	}
+
+	if err := keptnConfig.Validate(); err != nil {
+		return err
+	}
+
+	keptnCli, err := keptn.New(keptnConfig.URL, keptnConfig.Namespace, keptnConfig.Token.SecretRef.Name, nil)
+	if err != nil {
+		return fmt.Errorf("failed to create the keptn client %w", err)
+	}
+
+	shipyard, ok := data[keptn.ShipyardFileName]
+	if !ok {
+		return fmt.Errorf("shipyard.yaml not found")
+	}
+
+	appName := strings.ToLower(hr.Name + "-" + hr.Namespace)
+	if err := keptnCli.CreateOrUpdateProject(appName, shipyard); err != nil {
+		// if err := keptnCli.CreateProject(strings.ToLower("new-evaluation-project"), []byte(shipyard)); err != nil {
+		return err
+	}
+
+	if err := keptnCli.CreateService(appName, appName); err != nil {
+		return err
+	}
+
+	for k, v := range data {
```

Comment:
Can we use more specific names than `k` and `v`?

Reply:
Convention in Go states use single char variables but if it helps I can expand it to give more context (I will do that)